pkg/monitoring/metrics/virt-controller/vmsnapshot.go

```diff
@@ -0,0 +1,42 @@
+package virt_controller
+
+import (
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+
+	io_prometheus_client "github.com/prometheus/client_model/go"
+	snapshotv1 "kubevirt.io/api/snapshot/v1beta1"
+)
+
+var (
+	vmSnapshotMetrics = []operatormetrics.Metric{
+		VmSnapshotSucceededTimestamp,
+	}
+
+	VmSnapshotSucceededTimestamp = operatormetrics.NewGaugeVec(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_vmsnapshot_succeeded_timestamp_seconds",
+			Help: "Virtual Machine succeeded snapshots timestamp.",
+		},
+		[]string{"vm_name", "snapshot_name"},
+	)
+)
+
+func NewVMSnapshotCreated(snapshot *snapshotv1.VirtualMachineSnapshot) {
+	if snapshot.Status != nil &&
+		snapshot.Status.Phase == snapshotv1.Succeeded &&
+		snapshot.Status.CreationTime != nil {
```

Comment:
Just to make sure you are trying to update the time that the snapshot object was created and not the time that the snapshot was ready and completed (they should be pretty close like matter of seconds, but it is possible to be a couple of minutes in a cluster overload)

Reply:
I want the time it was created for snapshot that succeed

Reply:
@ShellyKa13 after consulting with Shirly We should be reporting the time when the snapshot is either succeeded or failed, not when the request was made..
We would also have another metric that will report the snapshot duration, but its in a different epic.

does `snapshot.CreationTimestamp` is the right timestamp?

Reply:
snapshot.CreationTimestamp this is k8s applying the creation of the CR. snapshot.status.CreationTime is the one reflecting when the snapshot was completed. For failed you will not have a timestamp of time that it failed snapshot.status.CreationTime will not get updated in failed case

Reply:
thanks! so I fixed it now to update the correct time.